### ISSUE TITLE
[YCQL] - A write heavy workload has stalled the cluster

### ISSUE THREAD

**Issue by jigar-biomatters** [YugaByte/yugabyte-db#1025]
I recently tried to insert some data into new tables created in a new keyspace, while a write heavy workload was writing tables in an existing keyspace.

This quickly stalled the cluster and now I am unable to load the master dashboard or read/write from/to any of the nodes. For about the past 18 hours, CPU utilisation has been constant on all the nodes and there appears to be some network activity between the nodes. There doesn't seem to be any memory or I/O bottleneck on the cluster.
There seems to be nothing significant in the logs (I checked all logs on a master+tserver and a tserver only node).

My YB cluster on AWS is as follows:
- 6 `c4.xlarge` nodes (each with 4 vCPUs and 7.5 GB of memory)
- this includes 3 masters and 6 tablet servers
- Each node has a 5 TB EBS volume associated with it (that provides 15000 IOPS per volume)
- yb version as previously reported by the dashboard is `version 1.1.9.0 build 18 revision 3ed9810e9a269e6f62ed553488f66fc217544c76 build_type RELEASE built at 06 Jan 2019 19:15:21 UTC`

I wonder what is happening and how to recover from this. More importantly, I want to try and figure out why this happened because this would definitely not be ideal in a production use case.

**Comment by kmuthukk** [comment]
@jigar-biomatters - in the http://master:7000/tablet-servers view, how many tablets do you have on each yb-tserver?

**Comment by kmuthukk** [comment]
happy to take a look at the cluster/logs when you have time. For clusters with larger number of tablets, in 1.2 release, we have made some improvements to require much lesser memory. Not sure if you are hitting that limit on this setup (which has about 7.5GB) per node.

**Comment by kmuthukk** [comment]
Would  be good to see the `ulimit -a` values from one of the nodes (assuming all nodes are identical).

**Comment by jigar-biomatters** [comment]
@kmuthukk requests to http://master:7000/tablet-servers don't return at the moment (master replaced with an actual master IP address obviously).

Yes, all nodes are identical. `ulimit -a` is as follows on one of them:
```
core file size          (blocks, -c) 0
data seg size           (kbytes, -d) unlimited
scheduling priority             (-e) 0
file size               (blocks, -f) unlimited
pending signals                 (-i) 29810
max locked memory       (kbytes, -l) 64
max memory size         (kbytes, -m) unlimited
open files                      (-n) 1048576
pipe size            (512 bytes, -p) 8
POSIX message queues     (bytes, -q) 819200
real-time priority              (-r) 0
stack size              (kbytes, -s) 8192
cpu time               (seconds, -t) unlimited
max user processes              (-u) 29810
virtual memory          (kbytes, -v) unlimited
file locks                      (-x) unlimited
```
I'll ping you when I'm available so we can look at the cluster/logs.

**Comment by kmuthukk** [comment]
ulimit settings look fine.

**Comment by kmuthukk** [comment]
Looked at this with @jigar-biomatters, and it looked like it was an issue similar to  #822 where if requests time out when the system was overloaded, YB kept retrying them internally. For now, restarting the yb-tservers fixed the issue - but upgrading to 1.2.0 release (which has a fix for #822) should also address the issue.

@jigar-biomatters - closing this for now. But if it reappears, please do not hesitate to reopen the issue.
